File: Weld.Test/ViewModels/NestedViewModel.cs
using Weld.Attributes;

namespace Weld.Test.ViewModels
{
    [WeldViewModel]
    class NestedViewModel
    {
        public bool Enabled { get; set; }

        public string FirstName { get; set; }

        public SimpleViewModel Nested { get; set; }
        
    }
}

File: Weld.Test/ViewModels/NestedViewModel.ts
/// <reference path="../typings/jquery/jquery.d.ts" />
class NestedViewModel 
{
    static Enabled: string = "Enabled";
    static get $Enabled() : JQuery {
        return $("#Enabled");
    }
    static FirstName: string = "FirstName";
    static get $FirstName() : JQuery {
        return $("#FirstName");
    }
    static Nested_Enabled: string = "Nested_Enabled";
    static get $Nested_Enabled() : JQuery {
        return $("#Nested_Enabled");
    }
    static Nested_FirstName: string = "Nested_FirstName";
    static get $Nested_FirstName() : JQuery {
        return $("#Nested_FirstName");
    }
}